### DISCUSSION TITLE
programmierwoche 7: faker-library, Prjojektarbeit

### DISCUSSION THREAD

**Discussion by Wildeswiesel** [Wildeswiesel/Space-Waffel#9] [General]
Zu beginn haben wir Anfangs wieder ein kleines Quiz gemacht, diesen findet ihr [hier](https://forms.gle/LV4eYcwu2mGhNKBg9)

danach haben wir uns über functions in w3schools infomiert link finden sie [hier](https://www.w3schools.com/python/python_functions.asp)

dann haben wir eine Aufgabe bekommen

```
Generate 10 students (or 100 or 1000)
Show that you can access and print student information
Put all first names of students in a list
Use sets to find out whether there are duplicates in one line of code. 
If there are duplicates,  print out which ones are duplicated. For example if the list of first names is the following, it should print out the duplicates like this:
['John', 'Jane', 'John', 'Alice', 'Bob', 'Jane']
{'Jane': 2, 'John': 2}

```

Meine Lösung war wie folgt :

```
from faker import Faker
import random

fake = Faker()

students = []

for i in range(30):
    student = {
        "name": fake.name(),
        "age": random.randint(18, 25),
        "major": random.choice(["Computer Science", "Mathematics", "Physics"]),
        "gpa": round(random.uniform(2.0, 4.0), 2),
        "is_international": random.choice([True, False])
    }
    students.append(student)

name_counts = {}
for student in students:
    first_name = student["name"].split()[0]
    if first_name in name_counts:
        name_counts[first_name] += 1
    else:
        name_counts[first_name] = 1


for student in students:
    full_name = student["name"]
    first_name = full_name.split()[0]
    if name_counts[first_name] > 1:
        print(f"{first_name} {name_counts[first_name]}")
        name_counts[first_name] -= 1  
    else:
        print(first_name)


```

Problem war allerdings das das ergebnmis nicht wie oben raus kam sonder so:

```
Tina
David 3
Eric 2
Caitlin 2
Kenneth
Rachel
Julie
Tim
Tammy
Gabriel
Ashley
Mandy
Steven
Christopher
David 2
Caitlin
Adam
Angela
David
Robert 2
Margaret
Eric
Philip
Lori
Robert
Amber
John
```

als Lösung wurde uns dieser code gegeben:

```
from faker import Faker
import random

fake = Faker()

students = []

for i in range(100):
    student = {
        "name": fake.name(),
        "age": random.randint(18, 25),
        "major": random.choice(["Computer Science", "Mathematics", "Physics"]),
        "gpa": round(random.uniform(2.0, 4.0), 2),
        "is_international": random.choice([True, False])
    }
    students.append(student)

for student in students:
    full_name = student["name"]
    first_name = student["name"].split()[0]
    last_name = student["name"].split()[1]
    print(first_name, last_name)
    print("age:", student["age"])
    print("Major:", student["major"])
    print("gpa:", student["gpa"])
    print("is international student:", student["is_international"])
    print()


first_names = []
for student in students:
    full_name = student["name"]
    first_name = student["name"].split()[0]
    first_names.append(first_name)

print(first_names)


duplikate_count = {name: first_names.count(name)
                    for name in set(first_names) #'mit set sag ich das es keine duplicatres sein sollen'
                    if first_names.count(name) > 1}
print(duplikate_count)
```
Gegen Ende haben wir noch über die Grading Criteria gesprochen welche für unsere Projektarbeit relevant ist.

---

Diese Woche habe ich natürlich noch an unserem Projekt weitergearbeitet und habe mich in diesem Fall mit Flask auseinander gesetzt 

**Flask** ist ein in Python geschriebenes Webframework. Sein Fokus liegt auf Erweiterbarkeit und guter Dokumentation. Die einzigen Softwareabhängigkeiten (engl. dependency) sind Jinja2, eine Template-Engine, und Werkzeug, eine Softwarebibliothek zum Erstellen von WSGI-Anwendungen.

Habe dann auch erstmal verschiedene Websiten erstellt um zu schauen ob diese angezeigt werden und habe nach und nach verschiedene  Funktionen hinzugefügt 

Desweiteren habe ich mich noch mit SQLite auseinander gesetzt 

**SQLite** ist eine gemeinfreie Programmbibliothek, die ein relationales Datenbanksystem enthält. SQLite wird in Mobiltelefonen, in Browsern, Skype und vielen anderen Anwendungen eingesetzt. Es ist das verbreitetste und meistverwendete Datenbanksystem der Welt.

und habe dann auch schon versucht diese mit einzubinden und natürlich auch die Website dementsprechend konfiguriert (html, css, JS)
schwierigkeiten hatte ich
- die daten in die DB zu schrieben 
- aus der DB zu lesen und es mit der Website zu synchronsieren ohne manuell neu laden zu müssen
- meine Popup Seite wieder schließen
- einträge wieder zu löschen 

[Link](https://github.com/Tim10022023/ToDoListPythonProject) zu unserem GitHub Repo

**Comment by d4c4n** [comment]
Sehr cool. Mir gefällt eure Projektideee und ich bin schon gespannt, wie das Endresultat aussieht!

**Comment by SvenSrc** [comment]
Danke für die kurze Zusammenfassung von Montag. Inwiefern würdest du Flask in dein Projekt anwenden? Bzw. wie

**Reply by Wildeswiesel** [reply]
![image](https://github.com/Wildeswiesel/Space-Waffel/assets/152853691/2c3cfe8b-ce1b-4f70-8848-4767fa767f95)
also die programm struktur sieht immoment so aus, wo bei die app.py als start quasi dient und die anderen dann aufruft/rendert
die bedienung ist eigentlich ganz einfach - musste mir nur mal kurz anschauen wie ich die website aufrufen kann und so
